src/compiler/program.ts

```diff
@@ -2200,17 +2200,17 @@ namespace ts {
                 const emitHost = getEmitHost();
                 const emitFilesSeen = createMap<true>();
                 forEachEmittedFile(emitHost, (emitFileNames) => {
-                    verifyEmitFilePath(emitFileNames.jsFilePath, emitFilesSeen);
-                    verifyEmitFilePath(emitFileNames.declarationFilePath, emitFilesSeen);
+                    verifyEmitFilePath(emitFileNames.jsFilePath, emitFilesSeen, /*isJsFile*/ true);
+                    verifyEmitFilePath(emitFileNames.declarationFilePath, emitFilesSeen, /*isJsFile*/ false);
                 });
             }
 
             // Verify that all the emit files are unique and don't overwrite input files
-            function verifyEmitFilePath(emitFileName: string, emitFilesSeen: Map<true>) {
+            function verifyEmitFilePath(emitFileName: string, emitFilesSeen: Map<true>, isJsFile: boolean) {
                 if (emitFileName) {

```

Comment:
While you're here -- apparently at least one of `jsFilePath` and `declarationFilePath` is optional (or this `if` can be removed), but neither is marked that way in the definition of `interface EmitFileNames`.

Reply:
Marked `declarationFilePath` as optional since `forEachEmittedFile` sometimes assigns it `undefined`.